test/mktemp.js

```diff
@@ -0,0 +1,112 @@
+var shell = require('..'),
+    common = require('../src/common');
+
+var assert = require('assert'),
+    fs = require('fs'),
+    path = require('path');
+
+shell.config.silent = true;
+shell.rm('-rf', 'tmp');
+shell.mkdir('tmp');
+
+// Not testing chmod on Windows.
+var isNotWindows = common.platform !== 'win';
+var result;
+
+
+//
+// Invalids
+//
+
+shell.touch(path.resolve('.', 'tmp/tmp.A'));
+result = shell.mktemp(path.resolve('.', 'tmp/tmp.A')); // Guarenteed to have a confilct, no randomness.
+assert.ok(shell.error());
+assert.equal(result.code, 2);
+
+//
+// Valids
+//
+
+// Basics
+result = shell.mktemp();
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 1);
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isFile());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Directory
+result = shell.mktemp('-d');
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 1);
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isDirectory());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Custom Template
+result = shell.mktemp(path.resolve('.', 'tmp/tmp.XXX'));
```

Comment:
Why use `path.resolve()` here? Relative paths should be supported as well I think.